pandas/core/generic.py

```diff
@@ -66,6 +66,28 @@ def _single_replace(self, to_replace, method, inplace, limit):
     return result
 
 
+def _pipe(obj, func, *args, **kwargs):
+    """
+    Apply a function to a obj either by
+    passing the obj as the first argument
+    to the function or, in the case that
+    the func is a tuple, interpret the first
+    element of the tuple as a function and
+    pass the obj to that function as a keyword
+    arguemnt whose key is the value of the
+    second element of the tuple
+    """
+    if isinstance(func, tuple):
+        func, target = func
+        if target in kwargs:
```

Comment:
you can leave the ndframe/group u tests where they are 
I don't there are any standalone tests